src/flags/recursion.rs

```diff
@@ -95,27 +68,24 @@ impl Recursion {
     ///
     /// # Note
     ///
-    /// The configuration file's Yaml is read in any case, to be able to check for errors and print
-    /// out warnings.
+    /// If both configuration file and Args is error, this will return a Max-Uint value.
     ///
     /// # Errors
     ///
     /// If [depth_from_arg_matches](Recursion::depth_from_arg_matches) returns an [Error], this
     /// returns it.
     fn depth_from(matches: &ArgMatches, config: &Config) -> Result<usize, Error> {
-        let mut result: Result<usize, Error> = Ok(usize::max_value());
-
-        if config.has_yaml() {
-            if let Some(value) = Self::depth_from_config(config) {
-                result = Ok(value);
+        if let Some(recursion) = &config.recursion {
+            if let Some(depth) = recursion.depth {
+                return Ok(depth);
```

Comment:
This method breaks the previously agreed on precedence. Command line options should always override the config file options.